library/cohesity_oracle.py

```diff
@@ -0,0 +1,415 @@
+      - is protecting.
+    type: str
+    default: Physical
+    required: yes
+  force_register:
+    description:
+      - Enabling this option will force the registration of the Cohesity Protection Source.
+    type: bool
+    default: no
+
+extends_documentation_fragment:
+    - cohesity
+requirements: []
+'''
+
+EXAMPLES = '''
+# Register a Physical Cohesity Protection Source on a selected Linux endpoint using Defaults
+- cohesity_oracle:
+    server: cohesity.lab
+    username: admin
+    password: password
+    endpoint: mylinux.host.lab
+    state: present
+
+# Unegister an existing Cohesity Protection Source on a selected endpoint
+- cohesity_oracle:
+    server: cohesity.lab
+    username: admin
+    password: password
+    endpoint: myvcenter.host.lab
+    environment: VMware
+    state: absent
+'''
+
+RETURN = '''
+# Example return from a succesful registration of a Linux Physical Source
+{
+  "ProtectionSource": {
+    "hostType": "kLinux",
+    "id": {
+      "clusterId": 8621173906188849,
+      "clusterIncarnationId": 1538852526333,
+      "id": 240
+    },
+    "name": "10.2.55.72",
+    "type": "kHost"
+  },
+  "changed": true,
+  "item": "control",
+  "msg": "Registration of Cohesity Protection Source Complete"
+}
+
+# Example return from the succesful unregistration of a Protection Source
+{
+  "changed": true,
+  "id": 241,
+  "endpoint": "mylinux.host.lab"
+  "msg": "Unregistration of Cohesity Protection Source Complete"
+}
+'''
+
+
+class ProtectionException(Exception):
+    pass
+
+# => Determine if the Endpoint is presently registered to the Cohesity Cluster
+# => and if so, then return the Protection Source ID.
+
+
+def register_oracle_source(module, self, _id):
+    """
+    Register a physical source as a Oracle source.
+    """
+    server = module.params.get('cluster')
+    validate_certs = module.params.get('validate_certs')
+    token = self['token']
+    endpoint = self["endpoint"]
+    source_id = _id
+    db_user = module.params.get('db_username')
+    db_pwd = module.params.get('db_password')
+
+    try:
+        uri = "https://" + server + "/irisservices/api/v1/applicationSourceRegistration"
+        headers = {"Accept": "application/json",
+                   "Authorization": "Bearer " + token}
+        # Payload to register Oracle source.
+        payload = {
+          "appEnvVec": [
+            19
+          ],
+          "usesPersistentAgent": True,
+          "ownerEntity": {
+            "type": 6
+          }
+        }
+        payload["ownerEntity"]["id"] = source_id
+        payload["ownerEntity"]["displayName"] = endpoint
+        if db_user and db_pwd:
+            cred = dict(username=db_user, password=db_pwd)
+            payload["appCredentialsVec"] = list() 
+            payload["appCredentialsVec"].append(dict(credentials=cred, envType=19))
+        data = json.dumps(payload)
+        response = open_url(url=uri, data=data, headers=headers,
```

Comment:
can you please use management sdk, check cohesity_clone_vm.py, cohesity_policy.py

Reply:
Okay, will use SDK.